string type
uint32 uid
uint32[] log_uids